compiler/noirc_frontend/src/tests/visibility.rs

```diff
@@ -493,3 +493,28 @@ fn does_not_error_if_referring_to_top_level_private_module_via_crate() {
     "#;
     assert_no_errors(src);
 }
+
+#[test]
+fn visibility_bug_inside_comptime() {
```

Comment:
Pushed a test for point 1. I couldn't understand point 2. Could you provide the complete code that triggers that error?

Reply:
R.e. (2):
generating a function (`bar_get_foo_inner`) that accesses `Foo`'s private field (`foo_inner`) gives the warning:
```noir
mod foo {
    pub struct Foo {
        foo_inner: Field,
    }
}

use foo::Foo;

// warning: foo_inner is private and not visible from the current module
//   ┌─ src/main.nr:9:1
//   │
// 9 │ #[generate_inner_accessor]
//   │ -------------------------- foo_inner is private
//   │
#[generate_inner_accessor]
struct Bar {
    bar_inner: Foo,
}

comptime fn generate_inner_accessor(_s: StructDefinition) -> Quoted {
    quote {
        fn bar_get_foo_inner(x: Bar) -> Field {
            x.bar_inner.foo_inner
        }
    }
}

fn main(x: Bar) {
    let _ = bar_get_foo_inner(x);
}
```

Probably out of scope for this PR, but do we also want to warn/error on `x: Bar` as a parameter to `main`, since `Bar` contains `Foo`, which has a private member?

Reply:
> since Bar contains Foo, which has a private member?

I think that's fine? Or why would it error? As long as we don't access `foo_inner` I think it's not an issue.

Reply:
Done point 2.

Reply:
> I think that's fine? Or why would it error?

Idea being that we're breaking encapsulation of that private member because a user needs to input it when ABI encoding. If I add a new field then that's now a breaking change.

Reply:
Ooooh... I missed all the context surrounding main arguments. Thanks! I'll capture an issue for that.